### PR TITLE
Fixes #366: Added placement policy options to ovirt_vm

### LINKED ISSUE #366: Add the `placement_policy` option to the `ovirt_vm` resource
## Please describe what you would like to see

Add the `placement_policy` option to the `ovirt_vm` resource to allow for defining pinning to specific hosts.

### PR DESCRIPTION
**Pull request by janosdebugs**
## Please describe the change you are making

This PR adds placement policy options to the `ovirt_vm` resource as requested in #366.

## Are you the owner of the code you are sending in, or do you have permission of the owner?

Yes

## The code will be published under the BSD 3 clause license. Have you read and understood this license?

Yes

### CODE REVIEW

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Review by eslutsky — CHANGES_REQUESTED** [review]

File: ovirt/resource_ovirt_vm_test.go [hunk 1]
```diff
@@ -186,14 +188,81 @@ resource "ovirt_vm" "foo" {
...
+		hostIDs[i] = fmt.Sprintf("\"%s\"", host.ID())
+	}
+	config := fmt.Sprintf(
+		`
+provider "ovirt" {
+	mock = true
+}
+
+resource "ovirt_vm" "foo" {
+	cluster_id                = "%s"
```
[the remark this answers is not in the document]
Comment (janosdebugs) [reply ovirt/resource_ovirt_vm_test.go:214]: No it isn't?

File: ovirt/resource_ovirt_vm_test.go [hunk 2]
```diff
@@ -186,14 +188,81 @@ resource "ovirt_vm" "foo" {
...
+	clusterID := p.getTestHelper().GetClusterID()
+	templateID := p.getTestHelper().GetBlankTemplateID()
+
+	client := p.getTestHelper().GetClient().WithContext(context.Background())
+	hosts, err := client.ListHosts()
+	if err != nil {
+		t.Fatalf("Failed to list hosts (%v)", err)
+	}
+	hostIDs := make([]string, len(hosts))
+	for i, host := range hosts {
```
[the remark this answers is not in the document]
Comment (janosdebugs) [reply ovirt/resource_ovirt_vm_test.go:204]: We don't have an API call for hosts in only a single cluster.

File: ovirt/resource_ovirt_vm_test.go [hunk 3]
```diff
@@ -186,14 +188,81 @@ resource "ovirt_vm" "foo" {
...
+provider "ovirt" {
+	mock = true
+}
+
+resource "ovirt_vm" "foo" {
+	cluster_id                = "%s"
+	template_id               = "%s"
+    name                      = "test"
+    placement_policy_affinity = "migratable"
+    placement_policy_host_ids = [%s]
```
[the remark this answers is not in the document]
Comment (eslutsky) [reply ovirt/resource_ovirt_vm_test.go:218]: image

[the remark this answers is not in the document]
Comment (eslutsky) [reply ovirt/resource_ovirt_vm_test.go:218]: image

**Comment by ovirt-infra** [comment]
All tests passed

**Review by eslutsky — APPROVED** [review]

**Pull request merged**